singer/src/main/java/com/pinterest/singer/monitor/DefaultLogMonitor.java

```diff
@@ -184,6 +195,28 @@ public static LogMonitor getInstance(int monitorIntervalInSecs,
     return INSTANCE;
   }
 
+  // a log stream is considered inactive if:
+  // 1. decider is 0
+  // 2. instance sampling disables this logstream
+  protected boolean isLogStreamInactive(LogStream logStream) {
+    SingerLogConfig logConfig = logStream.getSingerLog().getSingerLogConfig();
+    String decider = logConfig.getLogDecider();
+    int deciderValue = Decider.getInstance().getDeciderValue(decider);
+    if (deciderValue == -1) {
+      // logstream is considered active if the decider doesn't exist
+      return false;
+    } else if (deciderValue == 0) {
+      // decider is 0
+      return true;
+    } else if (logConfig.getLogStreamProcessorConfig().getDeciderBasedSampling().equals(SamplingType.INSTANCE)) {
```

Comment:
Line 205 and 211 should be combined since they are dependent on each other from a logical standpoint. This is help make the code more readable.